server/scsynth/SC_GraphDef.cpp

```diff
@@ -835,7 +835,14 @@ inline uint32 BufColorAllocator::alloc(uint32 count)
 		outIndex = nextIndex++;
 	}
 	if (outIndex >= refsMaxSize) {
-		refs = (int16*)realloc(refs, refsMaxSize*2*sizeof(int16));
+		int16 * tmprefs = (int16*)realloc(refs, refsMaxSize*2*sizeof(int16));
+		if ( tmprefs == NULL ) {
+			free(refs);
+			refs = NULL;
+			throw std::runtime_error("buffer coloring error: reallocation failed.");
```

Comment:
i'm a little wary of this. do we know where/if this will be caught? do we know whether throwing here will leave the program in a valid state?

Reply:
I noticed similar throw statements in other methods (e.g. ReleaseInputBuffers or DoBufferColoring) and thought it was the standard way of handling allocation errors. 

If the realloc fails, the resulting pointer is null, and so the code will crash anyway :)

Reply:
Ah, you're right! Thanks.

Reply:
Most of the sclang codebase tends to use the error code paradigm, that's why I was skeptical at first.